var/spack/repos/builtin/packages/swak4foam/package.py

```diff
@@ -0,0 +1,210 @@
+        # Only call a build stage, not patch stage since it tries to discover
+        # the foam project directory based on the existence of file
+        # Build wrapper script
+        spec = self.spec
+        foam_spec = spec['openfoam']
+        foam_projdir = find_foam_projectdir(foam_spec)  # WM_PROJECT_DIR
+        foam_instdir = os.path.dirname(foam_projdir)    # WM_PROJECT_INST_DIR
+
+        script = join_path(self.stage.source_path, self.build_script)
+        with open(script, 'w') as out:
+            out.write(
+                """#!/bin/bash
+export FOAM_INST_DIR={foam_instdir}
+export WM_PROJECT_DIR={foam_projdir}
+. $WM_PROJECT_DIR/etc/bashrc ''  # No arguments
+
+# Local build (user appbin, libbin)
+. $PWD/local-userdir.sh  {user_dir}
+
+# Force bison to be found first
+bison_bin={bison_bin}
+PATH=$bison_bin:$PATH
+
+echo Build swak4foam with SPACK for $WM_PROJECT-$WM_PROJECT_VERSION
+echo WM_PROJECT_DIR = $WM_PROJECT_DIR
+
+./Allwmake  # No arguments
+#
+"""
+                .format(
+                    foam_instdir=foam_instdir,
+                    foam_projdir=foam_projdir,
+                    bison_bin=spec['bison'].prefix.bin,
+                    user_dir='$PWD/' + self.build_userdir))
+        set_executable(script)
+
+    def patch(self):
+        """Generate a swakConfiguration file.
+        Provide additional files or other patching.
+        """
+        spec = self.spec
+
+        # Copy files
+        assetdir = join_path(self.package_dir, 'assets')
+        for f in self.assets:
+            install(
+                join_path(assetdir, f),
+                join_path(self.stage.source_path, f))
+
+        # swakConfiguration file
+        config = join_path(self.stage.source_path, self.config_file)
+        with open(config, 'w') as out:
+            if '+python' in spec:
+                out.write(
+                    """# With python
+python_bin={python_bin}
+python_config="$python_bin/python-config"
+export SWAK_PYTHON_INCLUDE="$($python_config --cflags)"
+export SWAK_PYTHON_LINK="$($python_config --ldflags)"
+
+# Force python to be found first
+PATH=$python_bin:$PATH
+#
+""".format(python_bin=spec['python'].prefix.bin))
+            else:
+                out.write('# No python requested\n')
+
+        # Record the spack spec information
+        with open("log.spack-spec", 'w') as outfile:
+            outfile.write(spec.tree())
+
+        # Emit openfoam version
+        tty.info('Build for {0}'.format(spec['openfoam'].format('$_$@$%@+$+')))
+
+    def build(self, spec, prefix):
+        """Build using the OpenFOAM Allwmake script, with a wrapper to source
+        its environment first.
+        """
+        self.create_build_script()
+        args = []
+        if self.parallel:  # Build in parallel? - pass via the environment
+            os.environ['WM_NCOMPPROCS'] = str(self.make_jobs) \
+                if self.make_jobs else str(multiprocessing.cpu_count())
```

Comment:
I know you mentioned this before, but if I stringify self.make_jobs it can be either "None" or "". Both are really bad for WM_NCOMPROCS, which should either be unset (make in parallel) or have an integer number for a parallel build. There is no convenient 'everything' setting.
The code snippets:

    if [ -n "$WM_NCOMPPROCS" ]  ...
    if [ "$WM_NCOMPPROCS" -gt 1 -a -z "$MAKEFLAGS" ] ... 
    parOpt="-j $WM_NCOMPPROCS"  ...

However, if I do request `spack install -j 3 swak4foam`, it seems to be doing exactly the right thing and only using 3 procs.